core/src/main/java/org/apache/cxf/common/util/SortedArraySet.java

```diff
@@ -138,20 +142,21 @@ public boolean remove(Object o) {
             return false;
         }
         int idx = Arrays.binarySearch(tmp, o);
-        if (idx != -1) {
```

Comment:
The bug was here (idx == -1). Every negative value means "not found".

Reply:
👍